### DISCUSSION TITLE
VisualFBEditor 1.3.5

### DISCUSSION THREAD

**Discussion by XusinboyBekchanov** [XusinboyBekchanov/VisualFBEditor#554] [General]
- Added: Use WASM menu item
- Fixed: Find and Replace
- Added: Confirm first before Replace All in Project
- Fixed: Replacing in procedure
- Added: ByRef Designer As My.Sys.Object parameter to all events
- Added: "Create event handlers without static event handler if event allows it" option
- Added: IntelliSense for Types and Enums in blocks
- Fixed: Compiling Android Project with fbc 1.10.0
- Fixed: Crash IDE on close tab
- Added: SapiTTS Example
- Fixed: Compiling CanvasDraw Example in Linux
- Improved: Edit Control caret in Linux
- Added: Edit Control Font changing in Linux
- Added: Find Compilers in Linux
- Fixed: Compiling crash in Linux
- Fixed: Debugging with other debuggers in Linux
- Fixed: Opening empty files in Linux.
- Added: Saving the "Show tip of the day" setting when closing the Tip of the day, previously it was saved only when closing the IDE.
- Fixed: Changing Control bounds with mouse in Designer (Linux)
- Fixed: Checking Use Defines on Linux
- Fixed: Crashing IDE on Linux
- Fixed: Updating ToolBar and ToolPalette controls on Designer.
- Fixed: Application titles of Project templates
- Fixed: Compiling ScintillaControl with fbc 1.10.0
- Added: Module Numbering and Macro Numbering
- Fixed: Drop files with extensions .vfp, .vfs to IDE
- Added: Format With Basis Word menu item
- Fixed: Copy and Paste controls in Designer
- Fixed: Copy controls from other Form
- Improved: UnComment block
- Added: SQLite3 example
- Updated: app-release.apk
- Added: onDestroy event to Android activity
- Added: File not found messages when run Android applications
- Fixed: Showing Chinese characters in Chinese calendar example
- Improved: Collapsing and Uncollapsing constructions inside #if, #ifdef, #elseif and #else in Edit Control
- Changed: Showing Type procedures in Function List
- Fixed: Change Keywords Case with Old Symbol
- Improved: IntelliSense
- Removed: ByRef and ByVal parameters from unused identifier suggestions.
- Fixed: Compiling Bass Example
- Improved: Suggestions
- Fixed: Suggestions for not used Enum items
- Added: IntelliSense for Line Continuation
- Improved: Collapse of Lines with Line Continuation and Line Separator in Edit Control
- Fixed: Changing Location, Size Properties
- Added: Asm Keywords Help
- Added: Delete blank lines menu item
- Added: Convert selection code to Lowercase & Uppercase menu items
- Added: Canvas Draw Example
- Fixed: Crash IDE
- Fixed: Compiling with fbc 1.10.0
- Added: AutoSave after entered chars option.
- Fixed: Auto suggestions for Asm
- Improved: Asm IntelliSense
- Improved: ChineseCalendar Example

<hr /><em>This discussion was created from the release <a href='https://github.com/XusinboyBekchanov/VisualFBEditor/releases/tag/1.3.5'>VisualFBEditor 1.3.5</a>.</em>

**Comment by XusinboyBekchanov** [comment]
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/32607344/6ebf5541-f584-42db-810e-52a7539cfec6)

**Comment by XusinboyBekchanov** [comment]
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/32607344/ff16041e-952b-4b63-80c7-09a20e54c511)

**Comment by XusinboyBekchanov** [comment]
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/32607344/e0d75de5-7dbf-4983-9a3a-7ae597cf974d)

**Comment by alas82** [comment]
hi , it is a nice ide , but it doesn't fit with all screen's resolution ex look not tidy at all for FHD screens, please advise

**Reply by XusinboyBekchanov** [reply]
Hi, please show me a drawing that doesn't look good.

**Reply by alas82** [reply]
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/65008114/3f15c2cb-bd34-4c04-8a34-169209cab14e)
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/65008114/4ea70364-c707-4e22-8f4f-04ef5db2892b)

**Reply by XusinboyBekchanov** [reply]
According to the first picture:
Fixed: ReBar in Light mode:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/e1ecb907bb2f818a0c3991c4d680b72be389573d

According to the second picture:
Try the new, not yet released version. I don't see the form headers being repainted black in the designer.

**Reply by alas82** [reply]
thanks for the reply , i downloaded the latest copy but i dont know how to make it as exe if you can teach me , the problem the scale of the ide is not correct from screen to other

**Reply by XusinboyBekchanov** [reply]
>thanks for the reply , i downloaded the latest copy but i dont know how to make it as exe if you can teach me 

You can download and compile using this bat file:
https://github.com/XusinboyBekchanov/VisualFBEditor/blob/master/DownloadAndCompileWithQuestions.bat

>the problem the scale of the ide is not correct from screen to other

I'll look into this issue this weekend.

**Reply by XusinboyBekchanov** [reply]
>the problem the scale of the ide is not correct from screen to other

This has been fixed:

Fixed: High DPI:
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/8bed9265c90eaca33886db02414b1ee71907a83e

Fixed: High DPI:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/3f94ed8bdfbf87643191b0cc7e5ffdf196878851

**Comment by alas82** [comment]
>  You can download and compile using this bat file: https://github.com/XusinboyBekchanov/VisualFBEditor/blob/master/DownloadAndCompileWithQuestions.bat

 i tried all this many times it is not producing any exe, i even tried to compile it with the ide 1.3.5 but didn't work, anyway thanks for your replies

**Reply by XusinboyBekchanov** [reply]
When compiling with 1.3.5, you need to change "Controls\MyFbFramework" with the path of the latest MyFbFramework:
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/32607344/440de5b9-0aba-4eb6-bf60-9742b03db08c)

**Comment by alas82** [comment]
`22:21:21: Compilation: "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\fbc64.exe"  -b "VisualFBEditor.bas" -exx -v -x "../VisualFBEditor64.exe" -s gui "VisualFBEditor.rc" -i "C:\Users\Alaa_As\Downloads\VisualFBEditor.1.3.5\Controls\MyFbFramework" -i ".\Controls\MyFbFramework\inc" -p ".\Controls\MyFbFramework\Lib" -d _DebugWindow_=134500

22:21:21: FreeBASIC Compiler - Version 1.10.1 (2023-12-24), built for win64 (64bit)
22:21:21: Copyright (C) 2004-2023 The FreeBASIC development team.
22:21:21: standalone 
22:21:21: target :       win64, x86-64, 64bit
22:21:21: backend :      gcc
22:21:21: compiling :    VisualFBEditor.bas -o VisualFBEditor.c (main module)
C:\Users\Alaa_As\Downloads\VisualFBEditor\src\Main.bas(12) error 23: File not found, "mff/SearchBar.bi"

22:21:24: Found Errors (1 Pos)
22:21:24: Do not build file. Elapsed Time: 3.03 Seconds
22:21:24: Compilation: "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\fbc64.exe"  -b "mff.bi" -exx -v -x "../mff64.dll" -dll "mff.rc" -i "C:\Users\Alaa_As\Downloads\VisualFBEditor\Controls\MyFbFramework" -i ".\Controls\MyFbFramework\inc" -p ".\Controls\MyFbFramework\Lib" -d _DebugWindow_=134500

22:21:24: FreeBASIC Compiler - Version 1.10.1 (2023-12-24), built for win64 (64bit)
22:21:24: Copyright (C) 2004-2023 The FreeBASIC development team.
22:21:24: standalone 
22:21:24: target :       win64, x86-64, 64bit
22:21:24: backend :      gcc
22:21:24: compiling :    mff.bi -o mff.c (main module)
22:21:32: compiling C :  C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\bin\win64\gcc.exe -m64 -march=x86-64 -S -nostdlib -nostdinc -Wall -Wno-unused -Wno-main -Werror-implicit-function-declaration -O0 -fno-strict-aliasing -fno-ident -frounding-math -fno-math-errno -fwrapv -fno-exceptions -fno-asynchronous-unwind-tables -funwind-tables -Wno-format -masm=intel "mff.c" -o "mff.asm"
22:22:09: assembling :   C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\bin\win64\as.exe --64 --strip-local-absolute "mff.asm" -o "mff.o"
22:22:10: compiling rc :               C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\bin\win64\GoRC.exe /ni /nw /o /machine X64 /fo "mff.obj" "mff.rc"
22:22:10: linking :      C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\bin\win64\ld.exe -m i386pep -o "../mff64.dll" -subsystem console --dll --enable-stdcall-fixup -e DllMainCRTStartup -T "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\lib\win64\fbextra.x" --stack 2097152,2097152 --output-def "../mff64.def" -s -L ".\Controls\MyFbFramework\Lib" -L "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\lib\win64" -L "." "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\lib\win64\dllcrt2.o" "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\lib\win64\crtbegin.o" "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\lib\win64\fbrt0.o" "mff.o" "mff.obj" "-(" -lkernel32 -lgdi32 -lmsimg32 -luser32 -lversion -ladvapi32 -limm32 -lcomctl32 -lcomdlg32 -luuid -lole32 -loleaut32 -liphlpapi -lrasapi32 -lws2_32 -lshlwapi -lshell32 -lgdiplus -lddraw -ldxguid -luxtheme -lwinmm -lstrmiids -lwininet -lwinspool -lfbmt -lgcc -lmsvcrt -lmingw32 -lmingwex -lmoldname -lgcc_eh "-)" "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\lib\win64\crtend.o"
22:22:10: creating import library :    C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\bin\win64\dlltool.exe --def "../mff64.def" --dllname "mff64.dll" --output-lib "../libmff64.dll.a"

22:22:11: Found Errors (1 Pos)
22:22:11: Layout succeeded! Elapsed Time: 46.57 Seconds` theses what i am getting

**Reply by XusinboyBekchanov** [reply]
`22:21:21: Compilation: "C:\Users\Alaa_As\Downloads\FreeBASIC-1.10.1-winlibs-gcc-9.3.0\fbc64.exe" -b "VisualFBEditor.bas" -exx -v -x "../VisualFBEditor64.exe" -s gui "VisualFBEditor.rc" -i "C:\Users\Alaa_As\Downloads\VisualFBEditor.1.3.5\Controls\MyFbFramework"

I saw in the compilation line the old MyFbFramework (1.3.5), instead you should use the new version of MyFbFramework, because there is a SearchBar:
https://github.com/XusinboyBekchanov/MyFbFramework/blob/master/mff/SearchBar.bi

Maybe you have a new MyFbFramework located in this folder:
C:\Users\Alaa_As\Downloads\VisualFBEditor\Controls\MyFbFramework

Then you installed a new MyFbFramework to the MyFbFramework project itself, this is not needed. You must put the new MyFbFramework to the "VisualFBEditor project"

**Comment by hustbeef** [comment]
Now Curl is functioning properly with MFF. There is no Curl.exe in Windows 7. Should we update the ".bat" file with an executable file for easier downloading of the source code?

**Reply by XusinboyBekchanov** [reply]
Yes, you can update.
